lib/chef/resource/aws_elasticache_subnet_group.rb

```diff
@@ -0,0 +1,24 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+class Chef::Resource::AwsElasticacheSubnetGroup < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type AWS::ElastiCache
+
+  attribute :group_name, kind_of: String, name_attribute: true
+  attribute :description, kind_of: String, required: true
+  attribute :subnets,
+            kind_of: [ String, Array ],
+            required: true,
+            coerce: proc { |v| [v].flatten }
+
+  attribute :aws_tags # hack until bug is fixed https://github.com/chef/chef-provisioning-aws/issues/204
+
+  def aws_object
+    begin
+      driver.elasticache
+        .describe_cache_subnet_groups(cache_subnet_group_name: name)
+        .data[:cache_subnet_groups].first
+    rescue AWS::ElastiCache::Errors::CacheSubnetGroupNotFoundFault
+      nil
+    end
```

Comment:
The above seems totally insane, but from my understanding of the SDK this is the best I can come up with.